vault-core/src/main/java/org/apache/jackrabbit/vault/fs/impl/io/DocViewSAXImporter.java

```diff
@@ -1295,6 +1300,31 @@ public void endElement(String uri, String localName, String qName) throws SAXExc
         }
     }
 
+    private int getNumChildrenWithType(Node parent, Node child) throws RepositoryException {
+
+        int count=0;
+       
+        NodeIterator iter = parent.getNodes();
+
+        EffectiveNodeType ent = EffectiveNodeType.ofNode(parent); 
+        Optional<NodeDefinition> def = ent.getApplicableChildNodeDefinition(child.getName(), child.getPrimaryNodeType());
+        
+        while (iter.hasNext()) {
+            Node sibling = iter.nextNode();
+            Optional<NodeDefinition>  childDef = ent.getApplicableChildNodeDefinition(sibling.getName(), sibling.getPrimaryNodeType());
+            Boolean test = (def == childDef) || (def != null && def.equals(childDef));
+
+            /*This is not working
+            if(def.equals(childDef)) { 
+                count++; 
+            }*/
+            if(Arrays.equals(def.get().getRequiredPrimaryTypes(),childDef.get().getRequiredPrimaryTypes())) {
+                count++; 
+            }
```

Comment:
@kwin according to @ankitaagar, the `def.equals(childDef)` is never true. do you know why?

Reply:
Currently https://github.com/apache/jackrabbit-filevault/blob/master/vault-core/src/main/java/org/apache/jackrabbit/vault/util/EffectiveNodeType.java does not implement equals, hash and toString

Reply:
Added those methods now in https://github.com/apache/jackrabbit-filevault/commit/0087391e3e2dbe0f70fab5dcb1a283c354c13c25

Reply:
thanks. but @ankitaagar is comparing the `Optional<NodeDefinition>`, and not the effective node type.

Reply:
NodeDefinition includes the node name (https://docs.adobe.com/content/docs/en/spec/jsr170/javadocs/jcr-2.0/javax/jcr/nodetype/ItemDefinition.html) and therefore cannot ever be equal for two different (sibling) nodes.

Reply:
no. it's the name of the item definition, not the name of the child node. i.e. in this case it should be `*`
https://docs.adobe.com/content/docs/en/spec/jsr170/javadocs/jcr-2.0/javax/jcr/nodetype/ItemDefinition.html#getName()

Reply:
the problem is rather, that the [ItemDefintion](https://github.com/apache/jackrabbit-oak/blob/c6ddcc55bee3de915459af01e91edad32d538f3d/oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/nodetype/ItemDefinitionImpl.java#L42) doesn't implement equals :-(

@ankitaagar I think your must implement the equals yourself or we add it as helper function to `EffectiveNodeType`

Reply:
Right, but equals() is not implemented in https://github.com/apache/jackrabbit-oak/blob/trunk/oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/nodetype/ItemDefinitionImpl.java either (nor does it necessarily have to according to the JCR 2.0 spec). Feel free to raise a PR for Oak.